### ISSUE TITLE
Unable to build wgrib-2.0.8 on Expanse

### ISSUE THREAD

**Issue by climbfuji** [NOAA-EMC/hpc-stack#339]
**Describe the bug**

On Expanse with the following compilers, wgrib2-2.0.8 cannot be built:
```
export HPC_COMPILER="intel/19.1.1.217"
export HPC_MPI="intel-mpi/2019.8.254"
export HPC_PYTHON="python/3.8.5"
```
The YAML entry is:
```
wgrib2:
  build: YES
  version: 2.0.8
  install_as: 2.0.8
  lib: YES
  ipolates: 3
```
The error is:
```
jpc_enc.c: In function 'rateallocate':
jpc_enc.c:1674:14: error: 'DBL_MAX' undeclared (first use in this function); did you mean 'TMP_MAX'?
  mnrdslope = DBL_MAX;
              ^~~~~~~
              TMP_MAX
jpc_enc.c:1674:14: note: each undeclared identifier is reported only once for each function it appears in
make[4]: *** [Makefile:384: jpc_enc.lo] Error 1
make[4]: Leaving directory '/expanse/lustre/scratch/domh/temp_project/hpc-stack-20210929/src/hpc-stack-20210929-expat-test/pkg/wgrib2-2.0.8/jasper-1.900.1/src/libjasper/jpc'
make[3]: *** [Makefile:404: check-recursive] Error 1
make[3]: Leaving directory '/expanse/lustre/scratch/domh/temp_project/hpc-stack-20210929/src/hpc-stack-20210929-expat-test/pkg/wgrib2-2.0.8/jasper-1.900.1/src/libjasper'
make[2]: *** [Makefile:290: check-recursive] Error 1
make[2]: Leaving directory '/expanse/lustre/scratch/domh/temp_project/hpc-stack-20210929/src/hpc-stack-20210929-expat-test/pkg/wgrib2-2.0.8/jasper-1.900.1/src'
make[1]: *** [Makefile:319: check-recursive] Error 1
make[1]: Leaving directory '/expanse/lustre/scratch/domh/temp_project/hpc-stack-20210929/src/hpc-stack-20210929-expat-test/pkg/wgrib2-2.0.8/jasper-1.900.1'
make: *** [makefile:648: /expanse/lustre/scratch/domh/temp_project/hpc-stack-20210929/src/hpc-stack-20210929-expat-test/pkg/wgrib2-2.0.8/lib/libjasper.a] Error 2
BUILD FAIL!  Lib: wgrib2 Error:2
```

The solution would be to use a newer version of Jasper (the hpc-stack install of jasper-2.x.y works fine), but as far as I understand wgrib2 requires jasper-1.900.1. The reason for this error is a missing `include float.h` statement in `jpc_enc.c`, which is apparently required for some newer operating systems and which was added in jasper-2.x.y .

**To Reproduce**

Check out develop and build as usual using `stack_noaa.yaml`.

**Expected behavior**

It would be nice - and required for some applications, but not right now for just the ufs-weather-model - to be able to build wgrib2.

**System:**

Xsede Expanse, see details above

**Comment by kgerheiser** [comment]
This is an issue with wgrib2 itself. We have to get Wesley to upgrade to a newer version of Jasper to fix this.

**Comment by edwardhartnett** [comment]
Due to the brittle nature of the wgrib2 build, it's lack of public repository, and the fact there is no testing, we recommend that wgrib2 not be used for any additional operational processing. 

The NCELIBS-g2c library, which does the heavy lifting for wgrib2, may be used. It is being subjected to testing right now.

Please try to avoid using wgrib2 moving forward.

**Comment by kgerheiser** [comment]
@edwardhartnett the wgrib2 executable is heavily used for doing command line operations on grib2 files. Also, wgrib2 uses its own built in grib processing. G2C is an option in wgrib2 but it's used for testing.

**Comment by edwardhartnett** [comment]
We have requested that wgrib2 be brought into the modern era, but to no avail. And meanwhile it causes problem after problem. So we have to start moving away from it.

I have started adding tests to the g2c library and hope in time that can be extended to cover more of the functionality of wgrib2.

In the meantime, new uses of wgrib2 should be avoided, because that is just digging the hole deeper...
